server/src/grpc/api/mod.rs

```diff
@@ -0,0 +1,110 @@
+//! gRPC API parent module
+
+pub mod cancel_itinerary;
+pub mod confirm_itinerary;
+pub mod query_flight;
+
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::sync::Mutex;
+use svc_storage_client_grpc::prelude::flight_plan;
+
+/// Time to wait for a flight plan to be confirmed before cancelling it
+const ITINERARY_EXPIRATION_S: u64 = 30;
+
+lazy_static! {
+    static ref UNCONFIRMED_ITINERARIES: Mutex<HashMap<String, Vec<String>>> =
+        Mutex::new(HashMap::new());
+    static ref UNCONFIRMED_FLIGHT_PLANS: Mutex<HashMap<String, flight_plan::Data>> =
+        Mutex::new(HashMap::new());
+}
+
+/// gets a hashmap of unconfirmed itineraries
+/// "itineraries" are a list of flight plan IDs, which can represent DRAFT
+///  (in memory) or EXISTING (in database) flight plans
+pub fn unconfirmed_itineraries() -> &'static Mutex<HashMap<String, Vec<String>>> {
+    &UNCONFIRMED_ITINERARIES
+}
+
+/// gets a hashmap of unconfirmed flight plans
+pub fn unconfirmed_flight_plans() -> &'static Mutex<HashMap<String, flight_plan::Data>> {
+    &UNCONFIRMED_FLIGHT_PLANS
+}
+
+/// Gets itinerary from hash map of unconfirmed itineraries
+pub fn get_draft_itinerary_by_id(id: &str) -> Option<Vec<String>> {
+    unconfirmed_itineraries()
+        .lock()
+        .expect("Mutex Lock Error getting itinerary from temp storage")
```

Comment:
the .expect() function will panic if an `Err` is returned:
"Because this function may panic, its use is generally discouraged. Instead, prefer to use pattern matching and handle the [Err](vscode-file://vscode-app/usr/lib/code/out/vs/code/electron-sandbox/workbench/workbench.html) case explicitly, or call [unwrap_or], [unwrap_or_else], or [unwrap_or_default]."

In this case, the `Err` result should be handled and `None` should be returned by the function.
Alternatively, the function should not return a `Option<Vec<String>>` but `Result<Vec<string>, Error>` instead so that `Err` can be passed and handled by the calling function.

Reply:
https://github.com/Arrow-air/svc-scheduler/pull/39/commits/b98f89ac17fa9427aacb7e0021392d176fb42f10